pkg/cache/listers.go

```diff
@@ -0,0 +1,154 @@
+package cache
+
+import (
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+)
+
+// GenericLister is a lister skin on a generic Indexer
+type GenericLister interface {
```

Comment:
I'm wondering if we want this hierarchy (names TBD)?

GenericClusterLister
- List (across all clusters)
- ByCluster (returns a cache.GenericLister)

GenericLister
- List (across all namespaces)
- Get (should ideally only be for cluster-scoped things)
- ByNamespace (returns a GenericNamespaceLister)

GenericNamespaceLister
- List (everything in that ns)
- Get (by name in that ns)

Reply:
I've made these changes, I think those names do make more sense